File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Action/Utility/WaitAction.ts
import Async from "../../../../../Core/Async";
import AbstractGameAction, { ActionType } from "../AbstractGameAction";

export class WaitAction extends AbstractGameAction{
    Type = ActionType.WAIT;
    constructor(private dur:number) {
        super();
    }
    
    public async Update(): Promise<boolean> {
        await Async.Delay(this.dur);
        return true;
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Action/Utility/WaitAction.cs
using Cysharp.Threading.Tasks;
using ModTheSpire.Event;

namespace ModTheSpire.Effect.Other
{
    public class WaitAction : IGameAction
    {
        public float Duration { get; set; }
        public ActionType Type { get; }= ActionType.WAIT;
        private int mInterval { get; set; }
        public void Init()
        {
            throw new System.NotImplementedException();
        }

        public WaitAction(int duration)
        {
            mInterval = duration;
        }
        
        public async UniTask<bool> Update()
        {
            await UniTask.Delay(mInterval);
            return true;
        }
    }
}